### PR TITLE
[v1.13] CI IPsec backports

### PR DESCRIPTION
**Pull request by qmonnet**
Here's a bunch of IPsec CI backports labelled as `backport/author`, before the workflows diverge too much.

Should be similar to [the v1.14 equivalent](https://github.com/cilium/cilium/pull/29966), although I needed to backport two preparatory commits to make `kind.sh` behave.

 * [x] #25317 (@chancez)
 * [ ] #26338 (@ysksuzuki)
 * [x] #29485 (@brb)
 * [x] #29514 (@brb)
 * [x] #29793 (@qmonnet)

Once this PR is merged, you can update the PR labels via:
```upstream-prs
for pr in 25317 26338 29485 29514 29793; do contrib/backporting/set-labels.py $pr done 1.13; done
```
or with
```
make add-labels BRANCH=v1.13 ISSUES=25317,26338,29485,29514,29793
```

### CODE REVIEW

**Comment by qmonnet** [comment]
/test-backport-1.13

**Comment by qmonnet** [comment]
`kind.sh` is not called correctly, it accepts fewer arguments on v1.13 - I rejected them when addressing the conflict, probably shouldn't have. I guess I'm missing one or two commits in the list. Turning this into draft for now.

**Comment by qmonnet** [comment]
/test-backport-1.13

**Comment by qmonnet** [comment]
/test-backport-1.13

**Comment by qmonnet** [comment]
/test-backport-1.13

**Comment by qmonnet** [comment]
/test-backport-1.13

**Review by chancez — APPROVED** [review]

**Comment by qmonnet** [comment]
/test-1.24-4.19

**Comment by qmonnet** [comment]
@jschwinger233 Would you mind taking a look for Martynas' commits, please?

**Comment by qmonnet** [comment]
/test-backport-1.13

**Review by jschwinger233 — APPROVED** [review]
brb's commits look good, thanks!

**Review by viktor-kurchenko — APPROVED** [review]

**Comment by qmonnet** [comment]
Thank you Viktor for reviewing for `ci-structure` and `github-sec`. :pray: 
Gray reviewed for Martynas' commits. :pray: 

The three failures in CI are expected, as mentioned above, now that I removed the test commit.

This PR is only missing TopHat's review, I'm marking as `ready-to-merge`.

**Comment by qmonnet** [comment]
Now needs a rebase

**Comment by qmonnet** [comment]
image

The three relevant workflows passed, I'm removing the test commit.

**Comment by qmonnet** [comment]
/test-backport-1.13

**Review by pchaigno — APPROVED** [review]
Approving for the two new commits. Thanks Quentin!

**Comment by qmonnet** [comment]
/test-1.21-4.19

**Comment by qmonnet** [comment]
The PR is ready to merge, once again. As mentioned before, the three failures are expected until the PR is merged and relevant Actions on `v1.13` get updated.

File: .github/workflows/tests-ipsec-upgrade.yaml
Comment (pchaigno) [inline .github/workflows/tests-ipsec-upgrade.yaml:338]: That doesn't look correct actually. We were skipping only Missed tail calls and now we'll skip all packet drops. We should use `--expected-drop-reasons` instead. No sure where my head was at.

Comment (qmonnet) [reply .github/workflows/tests-ipsec-upgrade.yaml:338]: @pchaigno Ouch, yeah I see what you mean. My bad. Are you working on a fix? If not, I can do it today (and for https://github.com/cilium/cilium/pull/30211).

Comment (pchaigno) [reply .github/workflows/tests-ipsec-upgrade.yaml:338]: I'm not. Go for it :slightly_smiling_face:

Comment (rgo3) [reply .github/workflows/tests-ipsec-upgrade.yaml:338]: I just found this PR while looking at how to add `--expected-drop-reasons "+Failed to update or lookup TC buffer"`. Should `!no-missed-tail-calls` also come back?

Comment (pchaigno) [reply .github/workflows/tests-ipsec-upgrade.yaml:338]: Nop, `no-missed-tail-calls` doesn't exist anymore (unless I messed up).

Comment (rgo3) [reply .github/workflows/tests-ipsec-upgrade.yaml:338]: yeah I didn't ask the right question. I meant if I should also add missed tail calls as an expected drop reason? I assume we still need to skip those failures.

Comment (qmonnet) [reply .github/workflows/tests-ipsec-upgrade.yaml:338]: (I suppose we need `--expected-drop-reasons=+Missed tail call`?)
[EDIT - Sorry, my page didn't refresh and I didn't see the rest of this conversation below my earlier message, until now :facepalm:]

Comment (pchaigno) [reply .github/workflows/tests-ipsec-upgrade.yaml:338]: Ah, yes. We need both Missed tail calls and CT buffer drops to be allowlisted.

Comment (qmonnet) [reply .github/workflows/tests-ipsec-upgrade.yaml:338]: I see Robin has this covered in https://github.com/cilium/cilium/pull/30249

**Pull request merged**
